pkg/volume/util/atomic_writer.go

```diff
@@ -414,6 +414,7 @@ func (w *AtomicWriter) writePayloadToDir(payload map[string]FileProjection, dir
 // baz -> ..data/baz
 func (w *AtomicWriter) createUserVisibleFiles(payload map[string]FileProjection) error {
 	for userVisiblePath := range payload {
+		userVisiblePath = filepath.Clean(userVisiblePath)
```

Comment:
Can you explain how this solves the windows issue?  It is not clear to me from the tests.

Reply:
This test is only for linux, so it's not clear for windows cases.

For windows containers, users may set path to `foo/bar`, while `filepath.Clean()` will convert it to `foo\bar`. The targetDir has already cleaned up to something like `c:\var\lib\kubelet...`, so following logic will work now.

Reply:
In the original bug report, the sample showed a container command that used Linux style path separators.  Even if this is fixed, will the user still have to change their program?

Reply:
Also should a windows test also be added?

Reply:
The unit tests on Windows are not running because of symlinks inside vendors. We may add this later if that could be resolved. @msau42 WDYT?

Reply:
Ok sounds fine. I will do a final review soon

Reply:
@msau42 Thanks for reminding this. validatePayload() uses path.Clean(), but it doesn't convert slash to platform's Separator. It's reasonable to fix the issue there.